sunpy/map/mapbase.py

```diff
@@ -247,6 +247,8 @@ def __init__(self, data, header, plot_settings=None, **kwargs):
             # Put import here to reduce sunpy.map import time
             from matplotlib import colors
             self.plot_settings['norm'] = colors.Normalize()
+            self.norm_to_check_contour = self.plot_settings['norm']
```

Comment:
Is this used somewhere?

Reply:
```
  if contour_args['norm'] == self.norm_to_check_contour:
      contour_args['norm'] = None
```
I expected this if condition to evaluate to True, but it is not (as the two objects are different). Do you have any idea why this might be happening?

Reply:
How are they difference exactly? Difference values?

Reply:
For eg like this `<matplotlib.colors.Normalize object at 0x7edef5363520>`
`<matplotlib.colors.Normalize object at 0x7edef53c3550>`

Reply:
Right, so the objects are different, so it won't work. 

What is the purpose of the check?

Reply:
So you want to set it to None if its been set at all?

Reply:
Set it to none if its been originally set in GenericMap

Reply:
Then can we not check if it exists?

Reply:
We can check `isinstance(norm , colors.Normalize())` but the problem with is this that mlp contour accepts string as well as `colors.Normalize()` for norm
https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.contour.html

Reply:
What if we check if just exists?

Reply:
Nope, if we just check if that exits then user passing `norm` in setting or directly in draw_contour is useless

Reply:
Well if its set by a user directly into draw_contour, cant we just use that and ignore if it exists in plot settings?

Reply:
> Well if its set by a user directly into draw_contour, cant we just use that and ignore if it exists in plot settings?

Ah, Thanks for pointing this out, fixed this in new commit

Reply:
So is this line still needed?

Reply:
Still it's a issue if passed through plotsettings (commit I made fixes it for draw_contour only)

Reply:
Yes we can, adding a comment about it.